backend/module/proxy/proxy.go

```diff
@@ -35,12 +36,17 @@ func New(cfg *any.Any, log *zap.Logger, scope tally.Scope) (module.Module, error
 		return nil, err
 	}
 
-	// Validate that each services constructs a parsable URL
 	for _, service := range config.Services {
 		for _, ar := range service.AllowedRequests {
-			_, err := url.Parse(fmt.Sprintf("%s%s", service.Host, ar.Path))
-			if err != nil {
-				return nil, fmt.Errorf("unable to parse the configured URL for service [%s]", service.Name)
+			switch t := ar.PathType.(type) {
+			case *proxyv1cfg.AllowRequest_Path:
+				// For exact path type, validate that each services constructs a parsable URL
+				_, err := url.Parse(fmt.Sprintf("%s%s", service.Host, t.Path))
```

Comment:
decided to just do this validation when it's an exact path type as a path regex might contain characters like `^` which would fail `url.Parse`